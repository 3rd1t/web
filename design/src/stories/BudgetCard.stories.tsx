import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
