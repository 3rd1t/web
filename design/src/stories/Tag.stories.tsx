import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
