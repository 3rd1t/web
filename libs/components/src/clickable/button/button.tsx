import { prependListener } from "process";
import React from "react";
import { Base } from "../base/base";

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string;
  textColor?: string;
  bgColor?: string;
  onClick: () => void;
  iconLeft?: React.ReactNode;
  iconRight?: React.ReactNode;
  className?: string;
}

export const Button = (props: ButtonProps) => {
  const className = [
    "focus:outline-none transition duration-200 rounded-sm py-1 px-4 font-medium",
    props.textColor,
    props.bgColor ||
      "bg-gradient-to-tr from-indigo-900 to-purple-800 hover:from-indigo-800 hover:to-purple-700",
    props.className,
  ].join(" ");

  return (
    <button className={className} onClick={props.onClick}>
      <Base
        label={props.label}
        iconLeft={props.iconLeft}
        iconRight={props.iconRight}
      ></Base>
    </button>
  );
};

export default Button;
