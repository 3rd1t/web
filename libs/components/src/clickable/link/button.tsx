import React from "react";
import { Link } from "./link";

/* eslint-disable-next-line */
export interface ButtonProps {
  label?: string;
  textColor?: string;
  bgColor?: string;
  href: string;
  iconLeft?: React.ReactNode;
  iconRight?: React.ReactNode;
  className?: string;
}

export const Button = (props: ButtonProps) => {
  return (
    <Link
      label={props.label}
      textColor={props.textColor}
      bgColor={
        props.bgColor ||
        "bg-gradient-to-t from-indigo-900 to-purple-800 hover:from-indigo-800 hover:to-purple-700"
      }
      href={props.href}
      iconLeft={props.iconLeft}
      iconRight={props.iconRight}
      className={props.className}
    ></Link>
  );
};

export default Button;
