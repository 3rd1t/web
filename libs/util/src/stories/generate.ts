/**
 * Return all components from storybook.
 *
 * @example
 * import * as stories from "./<component>.stories"
 *
 * const testCases = generateTestCasesFromStories(stories)
 * testCases.forEach((s) => {
 *   it(s.name + "should match snapshot", () => {
 *     const { baseElement } = render(s.component())
 *     expect(baseElement).toMatchSnapshot()
 *   })
 * })
 *
 *
 * @param stories - import * as stories from "./<component>.stories"
 * @returns Name and render function for each component
 */
export function generateTestCasesFromStories(
  stories: Record<string, any>, // eslint-disable-line
): { name: string; component: () => HTMLElement }[] {
  return Object.entries(stories)
    .filter((s) => s[0] !== "default")
    .map((s) => {
      return {
        name: s[0],
        component: s[1],
      }
    })
}
