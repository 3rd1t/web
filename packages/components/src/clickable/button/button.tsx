import React from "react"
import { Base } from "../base/base"
import { getHoverColor } from "../util/hoverColor"

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string | React.ReactNode
  textColor: string
  bgColor?: string
  onClick: () => void
  iconLeft?: React.ReactNode
  iconRight?: React.ReactNode
  width?: string
}

export const Button = (props: ButtonProps) => {
  const classNames = [
    "focus:outline-none",
    props.width,
    props.textColor,
    getHoverColor(props.textColor),
    props.bgColor,
    getHoverColor(props.bgColor),
  ].join(" ")

  return (
    <button onClick={props.onClick} className={classNames}>
      <Base label={props.label} iconLeft={props.iconLeft} iconRight={props.iconRight}></Base>
    </button>
  )
}

export default Button
