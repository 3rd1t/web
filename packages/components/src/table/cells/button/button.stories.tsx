import { action } from "@storybook/addon-actions"

import React from "react"
import { Button, ButtonProps } from "./button"

export default {
  component: Button,
  title: "components/table/cell/Button",
  argTypes: { onClick: { action: "clicked" } },
}

export const primary = () => {
  /* eslint-disable-next-line */
  const props: ButtonProps = {
    label: "Label",
    onClick: action("button-click"),
  }

  return <Button label={props.label} onClick={props.onClick} />
}
