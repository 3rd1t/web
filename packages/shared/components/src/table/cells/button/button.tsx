import React from "react"
import Wrapper from "../wrapper/wrapper"

/* eslint-disable-next-line */
export interface ButtonProps {
  label: string
  onClick: () => void
}

export const Button = ({ label, onClick }: ButtonProps) => {
  return (
    <Wrapper>
      <button onClick={onClick} className="text-sm font-medium leading-5 text-indigo-600 hover:text-indigo-900">
        {label}
      </button>
    </Wrapper>
  )
}

export default Button
