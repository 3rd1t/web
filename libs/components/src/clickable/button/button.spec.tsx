import { render } from "@testing-library/react";
import * as stories from "./button.stories";
import { generateTestCasesFromStories } from "@perfolio/util/stories/generate";

describe("Button", () => {
  describe("Stories", () => {
    const testCases = generateTestCasesFromStories(stories);

    testCases.forEach((s) => {
      it(s.name + "should match snapshot", () => {
        const { baseElement } = render(s.component());
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
