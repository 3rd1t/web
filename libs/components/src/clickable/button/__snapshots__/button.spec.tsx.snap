// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Stories blackshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-gray-100 hover:text-gray-200 bg-gray-900 hover:bg-gray-800"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          Label
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories blockshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none w-full text-gray-100 hover:text-gray-200 bg-gray-900 hover:bg-gray-800"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          Label
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories iconOnlyshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-green-500 hover:text-green-600 bg-gray-200 hover:bg-gray-300"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories primaryshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-gray-900 hover:text-gray-800 bg-gray-200 hover:bg-gray-300"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          Label
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories redshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-red-100 hover:text-red-200 bg-red-600 hover:bg-red-500"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          Label
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories withIconLeftAndRightshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-gray-100 hover:text-gray-200 bg-gray-900 hover:bg-gray-800"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 21a4 4 0 01-4-4V5a2 2 0 012-2h4a2 2 0 012 2v12a4 4 0 01-4 4zm0 0h12a2 2 0 002-2v-4a2 2 0 00-2-2h-2.343M11 7.343l1.657-1.657a2 2 0 012.828 0l2.829 2.829a2 2 0 010 2.828l-8.486 8.485M7 17h.01"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
        <span>
          Label
        </span>
        <span>
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 9l-7 7-7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories withIconLeftshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-gray-100 hover:text-gray-200 bg-gray-900 hover:bg-gray-800"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 14l9-5-9-5-9 5 9 5z"
            />
            <path
              d="M12 14l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14z"
            />
            <path
              d="M12 14l9-5-9-5-9 5 9 5zm0 0l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14zm-4 6v-7.5l4-2.222"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
        <span>
          Label
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`Button Stories withIconRightshould match snapshot 1`] = `
<body>
  <div>
    <button
      class="focus:outline-none  text-gray-100 hover:text-gray-200 bg-gray-900 hover:bg-gray-800"
    >
      <div
        class="flex items-center justify-center px-4 py-2 space-x-2 rounded-sm"
      >
        <span>
          Label
        </span>
        <span>
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 5l7 7m0 0l-7 7m7-7H3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</body>
`;
